src/main/java/edu/umich/eecs/april/apriltag/ApriltagDetectorActivity.java

```diff
@@ -178,18 +307,18 @@ public boolean onOptionsItemSelected(MenuItem item) {
                 verifyPreferences();
                 Intent intent = new Intent();
                 intent.setClassName(this, "edu.umich.eecs.april.apriltag.SettingsActivity");
-                startActivity(intent);
+                startActivityForResult(intent, SETTINGS_ACTIVITY_CODE);
                 return true;
 
+            // TODO(john): Fix crash, possibly because detector is still running when preferences change.
```

Comment:
It's already patched around by disabling reset. This is a non-trivial refactor of the detector and I'd rather not do it in this PR

Reply:
But I believe you're introducing the crash. Before this PR, the camera is paused when entering settings and re-initialized after returning to the main view.

Reply:
It's not possible to pause the camera in the new framework. Rather, I tested removing the Analyzer, but it appears that it stops asynchronously.

For now, it would be better to handle settings reset from the settings page, so that the camera is always paused.

In the future, we can fix this by not using a global `static apriltag_detector_t` for state